### PR TITLE
Fix explain_format and auto_explain tests

### PR DESCRIPTION
**Pull request by whitehawk**
Fix explain_format and auto_explain tests

Problem:
'explain_format' test failed if 'optimizer' GUC was set to 'on'.
'auto_explain' test failed with any 'optimizer' GUC value.

Cause:
Explain command outputs values of GUCs that affect planning if their value
differ from the default. Default value for 'optimizer' GUC has been changed to
'off' recently. So, the output of the tests has been changed as well.

Fix:
The patch updates matchsubs in 'explain_format' test to exclude handling of
'optimizer' GUC value and introduces respective updates in the files with
expected output. For 'auto_explain' test, it just updates the files with
expected output.

### CODE REVIEW

File: src/test/regress/sql/explain_format.sql [hunk 1]
```diff
@@ -9,13 +9,13 @@
 -- m/^Settings:.*/
 -- s/^Settings:[,\s]*/Settings: /
 -- m/^Settings:.*/
--- s/,?\s*optimizer\w*\s*=\s*'off'//g
+-- s/,?\s*optimizer\w*\s*=\s*'on'//g
```
Comment (bandetto) [inline src/test/regress/sql/explain_format.sql]: Maybe ignore `off` too?
```suggestion
-- s/,?\s*optimizer\w*\s*=\s*'(on|off)'//g
```

File: src/test/regress/sql/explain_format.sql [hunk 2]
```diff
@@ -9,13 +9,13 @@
...
 -- end_matchsubs
 
 -- ignore variable JIT gucs which can be shown when SETTINGS=ON
 -- start_matchignore
 -- m/^\s+jit\w*:/
--- m/\s*optimizer:\s*"off"/
+-- m/\s*optimizer:\s*"on"/
```
Comment (bandetto) [inline src/test/regress/sql/explain_format.sql]: Why are these matchignores needed? Don't we catch everything with matchsubs above?

Comment (whitehawk) [reply src/test/regress/sql/explain_format.sql]: It covers cases like:
```diff
@@ -342,6 +342,7 @@
   Optimizer: "GPORCA"
   Settings: 
     cpu_index_tuple_cost: "N.N"
+    optimizer: "on"
     random_page_cost: "N"
 (1 row)
 select explain_filter('EXPLAIN (FORMAT YAML, VERBOSE, SETTINGS ON) SELECT * from boxes;');
@@ -384,6 +385,7 @@
   Settings: 
     cpu_index_tuple_cost: "N.N"
     random_page_cost: "N"
+    optimizer: "on"
 (1 row)
```

File: src/test/regress/sql/explain_format.sql (hunk 1 above)
Comment (whitehawk) [reply src/test/regress/sql/explain_format.sql]: I think we shouldn't try to catch 'off', as it is the default value now, and we expect that it shouldn't be printed in verbose mode.

**Review by red1452 — COMMENTED** [review]
Why do we ignore optimizer "on" or "off" output, if GUC "optimizer" is not set in the test? I think it is better to remove match subs and check explain's output with the optimizer GUC's value.

**Comment by whitehawk** [comment]
> Why do we ignore optimizer "on" or "off" output, if GUC "optimizer" is not set in the test? I think it is better to remove match subs and check explain's output with the optimizer GUC's value.

I'm wondering as well why it was originally done this way in the test. [The original commit](https://github.com/arenadata/gpdb/commit/7e86a6bb77eba846695f176d3ebe530f4a1b80a8) is not very informative about the 'optimizer' part.
I've removed the matchsubs for 'optimizer' completely.

**Review by red1452 — APPROVED** [review]
Update PR's description.

**Comment by whitehawk** [comment]
I've added fix for auto_explain test into this PR, as it failed due to the same reason.

**Review by bandetto — APPROVED** [review]

**Review by red1452 — APPROVED** [review]

**Pull request merged**